lib/AST/Decl.cpp

```diff
@@ -10333,9 +10333,10 @@ void MissingDecl::forEachMacroExpandedDecl(MacroExpandedDeclCallback callback) {
   auto *baseDecl = unexpandedMacro.baseDecl;
   if (!macroRef || !baseDecl)
     return;
+  auto *module = getModuleContext();
 
   baseDecl->visitAuxiliaryDecls([&](Decl *auxiliaryDecl) {
-    auto *sf = auxiliaryDecl->getInnermostDeclContext()->getParentSourceFile();
+    auto *sf = module->getSourceFileContainingLocation(auxiliaryDecl->getLoc());
```

Comment:
We probably have to check for an invalid source location here, because some auxiliary decls (e.g., those created for property wrappers) might have invalid source locations. Can we use this new scheme for valid source locations, but fall back to the old one otherwise?

Reply:
Thanks, done!